django/db/models/base.py

```diff
@@ -569,6 +570,15 @@ def _get_pk_val(self, meta=None):
         return getattr(self, meta.pk.attname)
 
     def _set_pk_val(self, value):
+        for parent_link, (parent, field) in zip(
+                self._meta.parents.values(),
+                self._meta.parents.items()
+                ):
+            if parent_link and field:
+                pk_ = parent_link.target_field
+                if isinstance(pk_, AutoField) and pk_.attname != 'id':
```

Comment:
I don't think that we need this check. I would rather remove from docs [note](https://github.com/django/django/blob/77d335e5abec889b15323975187a8d5b10bfcb0f/docs/topics/db/queries.txt#L965-L979) about setting `id` to `None`. That is outdated after this patch.

\cc @spookylukey

Reply:
Yes, just a small doubt:
I was under the impression that only `AutoField`'s were to be made `None`. So what would happen if I were to set a `pk` of a child model to some string value when it's parent model had  set a primary_key on any numerical Field? I would be unknowingly setting a string value to the `pk` of parent model when it was expecting a numerical value. Is this alright?

Reply:
> I was under the impression that only `AutoField`'s were to be made `None`.

You can also set `pk` that is an `AutoField` to a string value, in all such cases Django raises `ValueError`, so I don't see any issue in it. Moreover we can have a primary key that is not an `AutoField` but has a default value, e.g. `UUIDField(default=uuid.uuid4)` and this should also work.

Reply:
> You can also set `pk` that is an `AutoField` to a string value, in all such cases Django raises `ValueError`

Indeed. Thanks a lot.

Reply:
Actually, am a bit reluctant of the fact that updating the `pk` for a model will also update it for all its parent models. The [current docs](https://docs.djangoproject.com/en/3.0/ref/models/instances/#the-pk-property) don't seem to  tell the users regarding this effect. Shouldn't we at least update the docs accordingly? Or am I mistaken somewhere?